src/components/popover/Popover.tsx

```diff
@@ -0,0 +1,142 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { AlignemntHorizontal, AlignmentVertical } from './enums';
+import classes from './Popover.module.scss';
+
+import {
+  getHorizontalTranslate,
+  getLeftPosition,
+  getTopPosition,
+  getVerticalTranslate,
+  hasLeftMargin,
+  hasRightBottom,
+  hasRightMargin,
+  hasTopMargin,
+  preventOverflow
+} from './helpers';
+import { PopoverProps } from './interfaces';
+import { Portal } from '../shared';
+
+const Popover = ({
+  children,
+  open,
+  onClose,
+  anchorEl,
+  renderOpenEl,
+  edgeMarginUnit = 4,
+  safetyMarginUnit = 24,
+  anchorOriginVertical = AlignmentVertical.bottom,
+  anchorOriginHorisontal = AlignemntHorizontal.left,
+  transformOriginVertical = AlignmentVertical.top,
+  transformOriginHorizontal = AlignemntHorizontal.left
+}: PopoverProps) => {
+  const [internalOpen, setInternalOpen] = useState<boolean>(false);
+  const [anchorRects, setAnchorRects] = useState<DOMRect | null>(null);
+  const [contentRects, setContentRects] = useState<DOMRect | null>(null);
+  const [containerRects, setContainerRects] = useState<DOMRect | null>(null);
+
+  const renderElRef = useRef<HTMLDataElement | null>(null);
+  const contentRef = useRef<HTMLDivElement | null>(null);
+  const containertRef = useRef<HTMLDivElement | null>(null);
+
+  const isOpen = useMemo(() => (renderOpenEl ? internalOpen : open), [renderOpenEl, internalOpen, open]);
+
+  const anchorPosition = useMemo(() => {
+    const rects = anchorRects ? anchorRects : anchorEl?.current?.getBoundingClientRect();
+
+    return {
+      top: getTopPosition(rects, anchorOriginVertical),
+      left: getLeftPosition(rects, anchorOriginHorisontal)
+    };
+  }, [anchorEl, anchorOriginVertical, anchorOriginHorisontal, anchorRects]);
+
+  const originPosition = useMemo(() => {
```

Comment:
Here you can remove return keyword, as they are no code except return inside function